CreditManagement/models.py

```diff
@@ -100,73 +99,417 @@ def __str__(self):
+    objects = PendingDiningTransactionManager()
+
     class Meta:
-        proxy = True
+        managed = False
+
+    @classmethod
+    def get_all_transactions(cls, user=None, association=None):
+        if association:
+            # Return none, no associations can be involved in dining lists
+            return cls.objects.none()
+        return cls.objects.get_queryset(user=user)
+
+    @classmethod
+    def get_user_balance(cls, user):
+        return cls.objects.all().compute_user_balance(user)
+
+    @classmethod
+    def get_association_balance(cls, association):
+        return cls.objects.all().compute_association_balance(association)
+
+    @classmethod
+    def annotate_balance(cls, users=None, associations=None, output_name=balance_annotation_name):
+        if associations:
+            return cls.objects.annotate_association_balance(associations, output_name=cls.balance_annotation_name)
+        else:
+            return cls.objects.annotate_users_balance(users, output_name=cls.balance_annotation_name)
+
+    @classmethod
+    def get_association_credit(cls, association):
+        """
+        Compute the balance according to this model based on the given association
+        :param association: The association
+        :return: The balance in Decimal
+        """
+        return Decimal(0.00)
 
-    objects = AssociationCreditQuerySet.as_manager()
+    def finalise(self):
+        raise NotImplementedError("PendingDiningTransactions are read-only")
 
-    def get_balance(self):
-        query = AssociationWithCredit.objects.filter(pk=self.pk).annotate_balance()
-        return query[0].balance
\ No newline at end of file
+    def __get_fixedform__(self):
+        return FixedTransaction(source_user=self.source_user, source_association=self.source_association,
+                                target_user=self.target_user, target_association=self.target_association,
+                                amount=self.amount,
+                                order_moment=self.order_moment, description=self.description)
+
+
+class PendingDiningListTracker(models.Model):
+    """
+    Model to track all Dining Lists that are pending.
+    Used for creating Pending Dining Transactions
+    """
+    dining_list = models.OneToOneField(DiningList, on_delete=models.CASCADE)
+
+    objects = PendingDiningTrackerQuerySet.as_manager()
+
+    def finalise(self):
+        # Generate the initial list
+        transactions = []
+
+        # Get all corresponding dining transactions
+        # loop over all items and make the transactions
+        for dining_transaction in PendingDiningTransaction.objects.get_queryset(dining_list=self.dining_list):
+            transactions.append(dining_transaction.__get_fixedform__())
+
+        # Save the changes
+        with transaction.atomic():
+            for fixed_transaction in transactions:
+                fixed_transaction.save()
+            self.delete()
+
```

Comment:
The largest complexity of the model is the pendingdiningtransactions I agree, but the general outline of pending and fixed is relatively easy. AbstractTransaction (though not completely abstract in functionality) is the root of all transactions, any transaction anywhere trickles down to this position when queried. IT does this by retrieving the information (through the same methods) in its direct children. Which is AbstractPendingTransaction and FixedTransaction. AbstractPendingTransaction has a child named PendingTransaction - which is basically the same as FixedTransaction except for a different table name, (yes I could merge contents in another parent class but I didn't think it worth the time) - and PendingDiningTransactions. As said, this later is harder as the contents of this table is retrieved through a query to avoid information duplication and the subsequent problems that could occur (i.e. the complex code when a user signs in/out, adjusts any (futur) payment option, whether it be auto_pay, association_pays_kitchen_cost or whatever people want in the future that in turn requires all made transactions to be altered to represent the new values).

As for the cyclic dependency you mention below. There actually isn't. It's the credit system that has a dependency on the 'Dining' app now. Yes, there is a fragment still in dining forms, but it's of the old style and is no longer needed, just hadn't removed it yet.

Reply:
> As for the cyclic dependency you mention below. There actually isn't. It's the credit system that has a dependency on the 'Dining' app now.

:+1: :ok_hand: